core/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroUtils.java

```diff
@@ -15,25 +15,70 @@
 
 package io.confluent.kafka.schemaregistry.avro;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import net.thisptr.jackson.jq.BuiltinFunctionLoader;
+import net.thisptr.jackson.jq.JsonQuery;
+import net.thisptr.jackson.jq.Scope;
+import net.thisptr.jackson.jq.Versions;
+import org.apache.avro.AvroTypeException;
 import org.apache.avro.Schema;
 import org.apache.avro.SchemaParseException;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class AvroUtils {
 
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
   /**
    * Convert a schema string into a schema object and a canonical schema string.
    *
    * @return A schema object and a canonical representation of the schema string. Return null if
    *     there is any parsing error.
    */
   public static AvroSchema parseSchema(String schemaString) {
+    Schema schema;
     try {
-      Schema.Parser parser1 = new Schema.Parser();
-      Schema schema = parser1.parse(schemaString);
-      //TODO: schema.toString() is not canonical (issue-28)
-      return new AvroSchema(schema, schema.toString());
+      try {
+        Schema.Parser parser1 = new Schema.Parser();
+        schema = parser1.parse(schemaString);
+      } catch (AvroTypeException e) {
+        // Must clear the parser
+        Schema.Parser parser1 = new Schema.Parser();
+        // In case of AvroTypeException,, attempt to replace quoted null with null for defaults
+        schema = parser1.parse(replaceQuotedNullWithNullForDefaults(schemaString));
+      }
     } catch (SchemaParseException e) {
       return null;
     }
+    return new AvroSchema(schema, schema.toString());
+  }
+
+  private static String replaceQuotedNullWithNullForDefaults(String schemaString) {
```

Comment:
It may be possible , we are requiring these type of hacks many times to work around customer issues. We  should  probably need a tool to always check customer configurations before rollout and have a channel to let customer know to fix their configuration.